src/engine/source/router/src/orchestrator.cpp

```diff
@@ -0,0 +1,548 @@
+    }
+    return std::nullopt;
+}
+
+void saveConfig(const std::weak_ptr<store::IStoreInternal>& wStore, const base::Name& storeName, const json::Json& dump)
+{
+    auto store = wStore.lock();
+    if (!store)
+    {
+        LOG_ERROR("Store is unavailable for dumping entries");
+    }
+    else
+    {
+        store->upsertInternalDoc(storeName, dump);
+    }
+}
+
+void Orchestrator::dumpTesters() const
+{
+    auto jDump = EntryConverter::toJsonArray(m_workers.front()->getTester()->getEntries());
+    saveConfig(m_wStore, m_storeTesterName, jDump);
+}
+
+void Orchestrator::dumpRouters() const
+{
+    auto jDump = EntryConverter::toJsonArray(m_workers.front()->getRouter()->getEntries());
+    saveConfig(m_wStore, m_storeRouterName, jDump);
+}
+/**************************************************************************
+ * Manage configuration - Loader
+ *************************************************************************/
+std::vector<EntryConverter> getEntriesFromStore(const std::shared_ptr<store::IStoreInternal>& store,
+                                                const base::Name& tableName)
+{
+    const auto jsonEntry = store->readInternalDoc(tableName);
+    if (base::isError(jsonEntry))
+    {
+        LOG_INFO("Router: {} table not found in store. Creating new table: {}",
+                 tableName.toStr(),
+                 base::getError(jsonEntry).message);
+        store->createInternalDoc(tableName, json::Json {"[]"});
+        return {};
+    }
+
+    auto json = base::getResponse(jsonEntry);
+    if (json.isEmpty())
+    {
+        LOG_WARNING("Router: {} table is empty", tableName.toStr());
+    }
+
+    return EntryConverter::fromJsonArray(json);
+}
+
+base::OptError Orchestrator::initWorker(const std::shared_ptr<IWorker>& worker,
+                                        const std::vector<EntryConverter>& routerEntries,
+                                        const std::vector<EntryConverter>& testerEntries)
+{
+    auto error = loadRouterOnWoker(routerEntries, worker);
+    auto error2 = loadTesterOnWorker(testerEntries, worker);
+
+    if (error && error2)
+    {
+        return base::Error {error->message + ". " + error2->message};
+    }
+    else if (error)
+    {
+        return error;
+    }
+    else if (error2)
+    {
+        return error2;
+    }
+
+    return std::nullopt;
+}
+
+// Public
+void Orchestrator::Options::validate() const
+{
+    if (m_numThreads < 1 || m_numThreads > 128)
+    {
+        throw std::runtime_error {"Configuration error: numThreads must be between 1 and 128"};
+    }
+    validatePointer(m_wStore, "store");
+    validatePointer(m_builder, "builder");
+    validatePointer(m_controllerMaker, "controllerMaker");
+    validatePointer(m_prodQueue, "prodQueue");
+    validatePointer(m_testQueue, "testQueue");
+    if (m_testTimeout < 1)
+    {
+        throw std::runtime_error {"Configuration error: testTimeout must be greater than 0"};
+    }
+}
```

Comment:
because it is called in the orchestrator construction or its configuration, and we want to stop the execution, since we cannot handle the error in here.